marimo/_dependencies/dependencies.py

```diff
@@ -1,203 +1,143 @@
 # Copyright 2024 Marimo. All rights reserved.
 from __future__ import annotations
 
+import importlib.metadata
 import importlib.util
+from dataclasses import dataclass
 
+from packaging import version
 
-class DependencyManager:
-    """Utilities for checking the status of dependencies."""
+from marimo import _loggers
 
-    @staticmethod
-    def require_pandas(why: str) -> None:
-        """
-        Raise an ModuleNotFoundError if pandas is not installed.
+LOGGER = _loggers.marimo_logger()
 
-        Args:
-            why: A string of the form "for <reason>" that will be appended
 
-        """
-        if not DependencyManager.has_pandas():
-            raise ModuleNotFoundError(
-                f"pandas is required {why}. "
-                + "You can install it with 'pip install pandas'"
-            ) from None
+@dataclass
+class Dependency:
+    pkg: str
+    min_version: str | None = None
+    max_version: str | None = None
 
-    @staticmethod
-    def require_polars(why: str) -> None:
-        """
-        Raise an ModuleNotFoundError if polars is not installed.
+    def has(self) -> bool:
+        """Return True if the dependency is installed."""
+        has_dep = importlib.util.find_spec(self.pkg) is not None
+        if not has_dep:
+            return False
 
-        Args:
-            why: A string of the form "for <reason>" that will be appended
+        if self.min_version or self.max_version:
+            self.warn_if_mismatch_version(self.min_version, self.max_version)
+        return True
 
-        """
-        if not DependencyManager.has_polars():
-            raise ModuleNotFoundError(
-                f"polars is required {why}. "
-                + "You can install it with 'pip install polars'"
-            ) from None
+    def has_at_version(
+        self, min_version: str | None, max_version: str | None = None
+    ) -> bool:
+        if not self.has():
+            return False
+        return _version_check(
+            pkg=self.pkg,
+            v=self.get_version(),
+            min_v=min_version,
+            max_v=max_version,
+        )
 
-    @staticmethod
-    def require_numpy(why: str) -> None:
+    def require(self, why: str) -> None:
```

Comment:
Should we also have a `require_at_version`?

Reply:
i can add it, but I dont think it is used yet